lib/Resource.js

```diff
@@ -141,6 +145,14 @@ Resource.prototype.relationships = function(relationName) {
   var self = this;
   var rawRelation = this._raw.relationships[relationName];
 
+  if (!self._base.id) {
+    rawRelation = this._raw.relationships[relationName] = {
```

Comment:
We need to alter both objects references to be the new object - this line is a `a = b = 6` where both `a` and `b` get set to `6` in one statement. I could break it out into two statement and shift the previous assignment around if you prefer?